### PR TITLE
Add batch writer

### PR DESCRIPTION
**Pull request by nathan-artie**
From https://github.com/artie-labs/custom-snapshots/pull/76

### CODE REVIEW

File: lib/jitter.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package lib
+
+import (
+	"math"
+	"math/rand"
+)
+
+func JitterMs(baseMs, maxMs, attempts int) int {
```
Comment (nathan-artie) [inline lib/jitter.go:8]: From https://github.com/artie-labs/custom-snapshots/blob/11fa65480b1942d5fa2867b5c27e24d4062612c7/lib/jitter.go#L13

File: lib/kafkalib/writer.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	"github.com/artie-labs/transfer/lib/cdc/util"
+	"github.com/artie-labs/transfer/lib/size"
+	"github.com/segmentio/kafka-go"
+
+	"github.com/artie-labs/reader/config"
+	"github.com/artie-labs/reader/lib"
+)
+
+const (
+	baseJitterMs = 300
```
Comment (nathan-artie) [inline lib/kafkalib/writer.go:18]: From https://github.com/artie-labs/custom-snapshots/blob/11fa65480b1942d5fa2867b5c27e24d4062612c7/lib/jitter.go#L13

Comment (nathan-artie) [reply lib/kafkalib/writer.go:18]: No other changes to this file other than removing `kafkalib` from function names.

File: lib/kafkalib/kafka.go [hunk 3]
```diff
@@ -72,23 +72,3 @@ func NewWriter(ctx context.Context, cfg config.Kafka) (*kafka.Writer, error) {
 
 	return writer, nil
 }
-
-type ReloadableWriter struct {
```
Comment (nathan-artie) [inline lib/kafkalib/kafka.go:76]: Moved to `writer.go`

**Review by Tang8330 — APPROVED** [review]

**Pull request merged**